File: DUWindow.py
#
#  DUWindow.py
#  DictUnifier
#
#  Created by Jiang Jiang on 3/7/10.
#  Copyright (c) 2010 Jjgod Jiang. All rights reserved.
#

import objc
from Foundation import *
from AppKit import *

class DUWindow(NSWindow):
    def awakeFromNib(self):
        self.registerForDraggedTypes_([NSFilenamesPboardType])

    def draggingEntered_(self, sender):
        pboard = sender.draggingPasteboard()
        types = pboard.types()
        opType = NSDragOperationNone
        if NSFilenamesPboardType in types:
            opType = NSDragOperationCopy
        return opType

    def performDragOperation_(self,sender):
        pboard = sender.draggingPasteboard()
        successful = False
        if NSFilenamesPboardType in pboard.types():
            files = pboard.propertyListForType_(NSFilenamesPboardType)
            print files
            successful = True
        return successful

File: DUWindow.m
//
//  DUWindow.m
//  DictUnifier
//
//  Created by Jjgod Jiang on 3/11/10.
//

#import "DUWindow.h"

@implementation DUWindow

- (void) awakeFromNib
{
    [self registerForDraggedTypes: [NSArray arrayWithObjects: NSPasteboardTypeFileURL, nil]];
}

- (NSDragOperation) draggingEntered: (id < NSDraggingInfo >) sender
{
    NSPasteboard *pboard = [sender draggingPasteboard];
    NSDragOperation opType = NSDragOperationNone;
    
    if ([[pboard types] containsObject: NSPasteboardTypeFileURL])
        opType = NSDragOperationCopy;
    
    return opType;
}

- (BOOL) performDragOperation: (id < NSDraggingInfo >) sender
{
    NSPasteboard *pboard = [sender draggingPasteboard];
    BOOL successful = NO;
    
    // 判断是否拖进单文件
    
    if (pboard.pasteboardItems.count <= 1) {
        //直接获取文件路径
        
        NSString *fileURL = [[NSURL URLFromPasteboard:pboard] path];
        [controller startConversion:fileURL];
        successful = NO;
    }
    
    
    return successful;
}

@end
